weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx

```diff
@@ -43,24 +46,67 @@ export const ChildPanelExportReport = ({
     null
   );
 
-  const selectedReportQuery = useQuery(GET_REPORT, {
+  const isNewReportSelected = isNewReportOption(selectedReport);
+
+  const [upsertReport, upsertReportResult] = useMutation(UPSERT_REPORT);
+  const reportQueryResult = useQuery(GET_REPORT, {
     variables: {id: selectedReport?.id ?? ''},
     skip: !selectedReport || isNewReportOption(selectedReport),
   });
 
-  const onAddPanel = () => {
-    const selectedReportDetails = selectedReportQuery.data?.view;
+  const isAddPanelDisabled =
+    !selectedEntity ||
+    !selectedReport ||
+    !selectedProject ||
+    reportQueryResult.loading ||
+    reportQueryResult.error != null ||
+    upsertReportResult.loading;
+
+  const onAddPanel = async () => {
+    if (isAddPanelDisabled) {
+      return;
+    }
+
     const slateNode = computeReportSlateNode(fullConfig, selectedPath);
-    // TODO - this will be replaced with correct add panel implementation later on
-    alert(`
-      Report id: ${selectedReport?.id}
-      Report name: ${selectedReport?.name}
-      Entity name: ${selectedEntity?.name}
-      Project name: ${selectedProject?.name}
-      Slate node: logged in dev console
-    `);
-    console.log('Selected report details:', selectedReportDetails);
-    console.log('Slate node to be added:', slateNode);
+    const entityName = selectedEntity.name;
+    const projectName = selectedProject.name;
+    let upsertBody;
+
+    if (isNewReportSelected) {
+      // https://wandb.atlassian.net/browse/WB-15495
+      alert('TODO: handle new report option');
+      return;
+    } else {
```

Comment:
**nitpick (non-blocking):** Since we return early, can remove `else`

Reply:
it won't be an early return once we actually add the new report logic!